File: leetCode/fukun/lc_066.js
const plusOne = function (digits) {
    let tmp = 1;
    let cursor = digits.length - 1;
    while (tmp) {
        if (cursor < 0) {
            digits.unshift(1);
            break;
        }
        if (digits[cursor] + tmp >= 10) {
            digits[cursor] = (digits[cursor] + tmp) % 10;
        } else {
            digits[cursor] += tmp;
            tmp--;
        }
        cursor--;
    }
    return digits;
};
console.log(plusOne([9]));

File: leetCode/XSS/lc_066.java
public class lc_066 {
    public static void main(String[] args) {
//加1，*
    }
    static class Solution {
        public int[] plusOne(int[] digits) {
            for(int i=digits.length-1;i>=0;i--)
            {
                if(digits[i]+1>=10)
                {
                    digits[i]=digits[i]+1-10;
                }
                else
                {
                    digits[i]=digits[i]+1;
                    return digits;
                }
            }
            int res[]=new int[digits.length+1];
            res[0]=1;res[1]=0;
            for(int i=1;i<digits.length;i++)
            {
                res[i+1]=digits[i];
            }
            return res;
        }
    }

}
